Dockerfile

```diff
@@ -1,30 +1,23 @@
-FROM debian:bullseye
+FROM debian:bookworm
 
 ENV DEBIAN_FRONTEND=noninteractive
 
 ADD dependencies.apt.txt ./
 
-RUN printf "deb-src http://deb.debian.org/debian bullseye main\ndeb-src http://deb.debian.org/debian-security/ bullseye-security main\ndeb-src http://deb.debian.org/debian bullseye-updates main" >> /etc/apt/sources.list
-
-RUN apt-get update && \
+RUN sed -i "s/Types: deb/Types: deb deb-src/" /etc/apt/sources.list.d/debian.sources && \
+    apt-get update && \
     xargs apt-get install --no-install-recommends -y < dependencies.apt.txt && \
     apt-get --no-install-recommends -y build-dep python3-apt && \
     apt-get clean && \
     rmdir /var/cache/apt/archives/partial
 
+# All requirements should be satisfied by dependencies.apt.txt – but feed it to
+# pip to keep us honest
 ADD requirements.txt ./
-
-RUN python3 -m pip install -r requirements.txt && \
+RUN sed -i 's/python-apt @ .*/python-apt/' requirements.txt && \
```

Comment:
Maybe it'd be better to filter python-apt from requirements.txt instead, python3-apt from Debian and the current source are both the same thing.

This replaces it with an outdated yanked version from pip

Reply:
No, it doesn't.

```
Requirement already satisfied: python-apt in /usr/lib/python3/dist-packages (from -r requirements.txt (line 9)) (2.6.0)
```

I couldn't find a way to make `pip` check that everything listed is installed without actually downloading anything, but that's what I actually wanted to do.

Reply:
I found a way to do this.

Reply:
Ah yea the one from apt is installed first